runtime/lua/vim/shared.lua

```diff
@@ -326,6 +326,21 @@ function vim.tbl_islist(t)
   end
 end
 
+-- Return the number of the value in table
```

Comment:
should explain how it differs from `#` . Also let's use the [lume.count docstring](https://github.com/rxi/lume#lumecountt--fn):

    --- Counts the number of values in table `t`.

The `fn` arg of `lume.count` looks useful. That tends to be the case for functional languages like Lua: a "visitor" function is often useful.